DevelopmentProposals/componentConnectInterface/componentConnectInterface.md

```diff
@@ -0,0 +1,62 @@
+## Project: Improved interface for connecting Components
+***OpenSim Development Proposal***
+
+### Overview
+Connecting Components currently looks something like "update the Input of my ConsoleReporter and connect it to this Output from my Model".
+Some users may find it more intuitive to (i) begin with the verb *connect*, and (ii) swap the order of the Input and Output.
+This project proposes to change the user interface for connecting Components while retaining all current functionality.
+See [issue #1118](https://github.com/opensim-org/opensim-core/issues/1118).
+
+### Requirements
+- No change to what can be connected or how the underlying connections are formed.
+- No redundancy in the interface (i.e., the new interface replaces the existing interface).
+
+### Architecture
+Changes to `Component`:
+- Create a new method for forming connections between Components. There will be several variants:
+  - `static void connect(const Object& object, Object& object)` and `static void connect(Object& object, const Object& object)`,
+    which will call [this method](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Common/ComponentConnector.h#L278).
+  - `static void connect(const AbstractOutput& output, AbstractInput& input, const std::string& annotation = "")` and
+    `static void connect(AbstractInput& input, const AbstractOutput& output, const std::string& annotation = "")`,
+    which will call [this method](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Common/ComponentConnector.h#L494).
+  - `static void connect(const AbstractChannel& channel, AbstractInput& input, const std::string& annotation = "")` and
+    `static void connect(AbstractInput& input, const AbstractChannel& channel, const std::string& annotation = "")`,
+    which will call [this method](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Common/ComponentConnector.h#L497).
+- Change the name of the existing `void connect(Component& root)` method to `finalizeConnections()`.
```

Comment:
I was unsure about `finalize` here but I think @tkuchida is trying to achieve something important. There are two distinct operations: (i) model-developers specifying what should be connected to what, and (ii) model-consumers telling their model to "find the things that the model-developer said you're supposed to be connected to". Right now, we use the word `connect()` for both of these operations, and that is confusing. I personally am fine with `finalizeConnections()` since it reduces the number of verbs in our interface, and the operation is similar to `finalizeFromProperties()`.

One option: remove this method altogether, and include "satisfying connections" as part of "finalizing from properties." This would be a larger project I think, though. We could either have a separate proposal for that, or have this proposal include multiple PRs.